src/engine/sync/enginesync.cpp

```diff
@@ -308,6 +308,10 @@ void EngineSync::activateMaster(Syncable* pSyncable) {
         qWarning() << "WARNING: Logic Error: Called activateMaster on a NULL Syncable.";
         return;
     }
+    if (m_pMasterSyncable == NULL){
```

Comment:
no, the current master syncable could already be null and that's ok.  Note that "pOldChannelMaster" is tested for null-ness below